keras_cv/models/feature_extractor/coca/coca_model.py

```diff
@@ -0,0 +1,244 @@
+                             f": {text_shape}")
+
+        text_dim = text_shape[1]
+        batch_size = images_shape[0]
+        if batch_size != text_shape[0]:
+            raise ValueError(f"Differing batch sizes between images and texts input. {batch_size} vs {text_shape[0]}")
+
+        # Build Layers
+        self.image_patching.build(images_shape)
+
+        # Add 1 for CLs token appended by patching
+        num_patches = (images_shape[1] // self.img_patch_size) * (images_shape[2] // self.img_patch_size) + 1
+        self.image_encoder.build((batch_size, self.encoder_width, num_patches))
+
+        text_shape_with_cls_token = [s for s in text_shape]
+        text_shape_with_cls_token[1] += 1
+        self.text_embedding.build(text_shape_with_cls_token)
+
+        self.unimodal_text_decoder.build(text_shape_with_cls_token)
+
+        self.contrastive_attn_pooling.build((batch_size, num_patches, self.encoder_width))
+        self.captioning_attn_pooling.build((batch_size, num_patches, self.encoder_width))
+
+        self.multimodal_text_decoder.build((batch_size, self.encoder_width, self.captioning_query_length),
+                                           text_shape_with_cls_token)
+
+        # Learnable Weights
+        self.cls_token = self.add_weight(shape=(batch_size, 1, text_dim), name="cls_token", trainable=True)
+
+        self.contrastive_query = self.add_weight(shape=(batch_size, self.encoder_width, self.contrastive_query_length),
+                                                 trainable=True)
+        self.captioning_query = self.add_weight(shape=(batch_size, self.encoder_width, self.captioning_query_length),
+                                                trainable=True)
+
+    def call(self, images, texts):
+        """
+        Forward pass of the Coca Model from raw image and text data
+
+        Args:
+            images: [batch_size, height, width, channels] representing images
+            texts: Tensor, typically represented as [batch_size, sequence_length, feature_length] or
+                [batch_size, sequence_length, num_heads, feature_length]. The sequence_length and/or feature_length
+                are required.
+
+        Returns:
+            Output: Output of the captioning Transformer Decoder with captioning cross-attention
+        """
+        img_encoding = self.image_patching(images) # [batch_size, encoder_width, img_patches_len+1]
+        img_encoding = self.image_encoder(img_encoding)  # [batch_size, img_patches_len+1, encoder_width]
+
+        # This is only needed for loss calculations
+        # contrastive_feature = self.con_attn_pooling(self.contrastive_query, img_encoding)
+
+        # [batch_size, encoder_width, captioning_query_length]
+        captioning_feature = self.captioning_attn_pooling(self.captioning_query, img_encoding)
+
+        # [batch_size, sequence_length+1, text_dim]
+        text_tokens = np.concatenate(texts, self.cls_token)
```

Comment:
Yeah, I considered that but I think because the token doesn't actually leave the model's intermediate representation, it's okay to leave here.

Otherwise, it would create a weird API where the input is of length N, but the output is length N-1.